pkg/kube/client/client.go

```diff
@@ -120,6 +143,21 @@ func configForProvider(ctx context.Context, local client.Client, providerConfigN
 					return nil, errors.Wrap(err, errInjectAzureCredentials)
 				}
 			}
+		case kconfig.IdentityTypeUpboundToken:
+			switch id.Source { //nolint:exhaustive
+			case xpv1.CredentialsSourceInjectedIdentity:
+				return nil, errors.Errorf("%s is not supported as identity source for identity type %s",
+					xpv1.CredentialsSourceInjectedIdentity, kconfig.IdentityTypeUpboundToken)
+			default:
+				tkn, err := resource.CommonCredentialExtractor(ctx, id.Source, b.local, id.CommonCredentialSelectors)
+				if err != nil {
+					return nil, errors.Wrap(err, errExtractUpboundCredentials)
+				}
+
+				if err := upbound.WrapRESTConfig(ctx, rc, string(tkn), b.store); err != nil {
```

Comment:
TODO: Trim whitespaces around the token. 

Thanks a lot @haarchri, for testing the PR and catching this 🙌